python/dataproc_templates/jdbc/jdbc_to_gcs.py

```diff
@@ -188,33 +195,60 @@ def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
             exit (1)
-        elif (partition_parameters == ""):
-            input_data=spark.read \
+        # Input is table name
+        elif (input_jdbc_table):
+            if (partition_parameters == ""):
+                input_data=spark.read \
+                    .format(constants.FORMAT_JDBC) \
+                    .option(constants.JDBC_URL, input_jdbc_url) \
+                    .option(constants.JDBC_DRIVER, input_jdbc_driver) \
+                    .option(constants.JDBC_TABLE, input_jdbc_table) \
+                    .option(constants.JDBC_NUMPARTITIONS, jdbc_numpartitions) \
+                    .option(constants.JDBC_FETCHSIZE, input_jdbc_fetchsize) \
+                    .load()
+            else:
+                input_data=spark.read \
                 .format(constants.FORMAT_JDBC) \
                 .option(constants.JDBC_URL, input_jdbc_url) \
                 .option(constants.JDBC_DRIVER, input_jdbc_driver) \
                 .option(constants.JDBC_TABLE, input_jdbc_table) \
+                .option(constants.JDBC_PARTITIONCOLUMN, input_jdbc_partitioncolumn) \
+                .option(constants.JDBC_LOWERBOUND, input_jdbc_lowerbound) \
+                .option(constants.JDBC_UPPERBOUND, input_jdbc_upperbound) \
                 .option(constants.JDBC_NUMPARTITIONS, jdbc_numpartitions) \
                 .option(constants.JDBC_FETCHSIZE, input_jdbc_fetchsize) \
                 .load()
-        else:
-            input_data=spark.read \
+        # Input is sql query
+        elif (input_jdbc_sql_query):
+            if (partition_parameters == ""):
+                input_data=spark.read \
+                    .format(constants.FORMAT_JDBC) \
+                    .option(constants.JDBC_URL, input_jdbc_url) \
+                    .option(constants.JDBC_DRIVER, input_jdbc_driver) \
+                    .option(constants.JDBC_QUERY, input_jdbc_sql_query) \
+                    .option(constants.JDBC_NUMPARTITIONS, jdbc_numpartitions) \
+                    .option(constants.JDBC_FETCHSIZE, input_jdbc_fetchsize) \
+                    .load()
+            else:
+                input_data=spark.read \
                 .format(constants.FORMAT_JDBC) \
                 .option(constants.JDBC_URL, input_jdbc_url) \
                 .option(constants.JDBC_DRIVER, input_jdbc_driver) \
-                .option(constants.JDBC_TABLE, input_jdbc_table) \
+                .option(constants.JDBC_QUERY, input_jdbc_sql_query) \
                 .option(constants.JDBC_PARTITIONCOLUMN, input_jdbc_partitioncolumn) \
                 .option(constants.JDBC_LOWERBOUND, input_jdbc_lowerbound) \
                 .option(constants.JDBC_UPPERBOUND, input_jdbc_upperbound) \
                 .option(constants.JDBC_NUMPARTITIONS, jdbc_numpartitions) \
                 .option(constants.JDBC_FETCHSIZE, input_jdbc_fetchsize) \
                 .load()
+        else:
```

Comment:
Can this be done neatly? May be using .options with a dict and adding properties that are required as per single level if-else statements.

Reply:
made changes